yapapi/strategy.py

```diff
@@ -25,6 +27,27 @@
 class MarketStrategy(DemandDecorator, abc.ABC):
     """Abstract market strategy."""
 
+    valid_prop_value_ranges: Dict[str, Tuple[Optional[float], Optional[float]]]
+
+    def set_valid_prop_value_ranges(
+        self, valid_prop_value_ranges: Dict[str, Tuple[Optional[float], Optional[float]]]
+    ) -> None:
+        self.valid_prop_value_ranges = valid_prop_value_ranges
+
+    async def answer_to_provider_offer(
+        self, our_demand: DemandBuilder, provider_offer: rest.market.OfferProposal
+    ) -> DemandBuilder:
+        updated_demand = deepcopy(our_demand)
+        for prop_name, valid_range in self.valid_prop_value_ranges.items():
+            prop_value = provider_offer.props.get(prop_name)
+            if prop_value:
+                if valid_range[0] is not None and prop_value < valid_range[0]:
+                    raise Exception(f"Negotiated property {prop_name} < {valid_range[0]}.")
```

Comment:
Maybe `ValueError` would be a better exception here?

Reply:
Fixed